common/autoware_universe_utils/include/autoware/universe_utils/ros/managed_transform_buffer.hpp

```diff
@@ -58,12 +58,69 @@ struct PairEqual
   }
 };
 using TFMap = std::unordered_map<Key, Eigen::Matrix4f, std::hash<Key>, PairEqual>;
+constexpr std::array<const char *, 3> warn_frames = {"map", "odom", "world"};
 
-class StaticTransformBuffer
+class ManagedTransformBuffer
 {
 public:
-  StaticTransformBuffer() = default;
+  explicit ManagedTransformBuffer(rclcpp::Node * node, const bool & has_static_tf_only)
+  : node_(node)
+  {
+    if (has_static_tf_only) {
+      get_transform_ = [this](
+                         const std::string & target_frame, const std::string & source_frame,
+                         Eigen::Matrix4f & eigen_transform) {
+        return getStaticTransform(target_frame, source_frame, eigen_transform);
+      };
+    } else {
+      tf_listener_ = std::make_unique<autoware::universe_utils::TransformListener>(node);
+      get_transform_ = [this](
+                         const std::string & target_frame, const std::string & source_frame,
+                         Eigen::Matrix4f & eigen_transform) {
+        return getDynamicTransform(target_frame, source_frame, eigen_transform);
+      };
+    }
+  }
+
+  bool getTransform(
+    const std::string & target_frame, const std::string & source_frame,
+    Eigen::Matrix4f & eigen_transform)
+  {
+    return get_transform_(target_frame, source_frame, eigen_transform);
```

Comment:
nit: this adds an extra level of indirection. the extra call and stack could be avoided by having the branch here (just ignore this if you do not agree)